### PR TITLE
[Storage]: Enable Hierarchical Namespace for storage account

### PR DESCRIPTION
**Pull request by Juliehzl**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
@Juliehzl  please help with this PR.

**Comment by Juliehzl** [comment]
@MyronFanQiu @qwordy Can you help with the PR?

File: src/azure-cli/azure/cli/command_modules/storage/_params.py [hunk 1]
```diff
@@ -154,6 +154,9 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
         c.argument('domain_guid', domain_guid_type)
         c.argument('domain_sid', domain_sid_type)
         c.argument('azure_storage_sid', azure_storage_sid_type)
+        c.argument('enable_hierarchical_namespace', options_list=['--enable-hierarchical-namespace', '--hns'],
+                   action='store_true', help=" Allow the blob service to exhibit filesystem semantics",
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/storage/_params.py]: file system

File: src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_account_scenarios.py [hunk 2]
```diff
@@ -169,6 +169,13 @@ def test_storage_create_default_sku(self, resource_group):
         create_cmd = 'az storage account create -n {} -g {}'.format(name, resource_group)
         self.cmd(create_cmd, checks=[JMESPathCheck('sku.name', 'Standard_RAGRS')])
 
+    @api_version_constraint(ResourceType.MGMT_STORAGE, min_api='2018-02-01')
+    @ResourceGroupPreparer(location='southcentralus')
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_account_scenarios.py]: Maybe you can add a name_prefix=

File: src/azure-cli/azure/cli/command_modules/storage/_params.py (hunk 1 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/storage/_params.py]: What's the type of it?

Comment (MyronFanQiu) [reply src/azure-cli/azure/cli/command_modules/storage/_params.py]: Can we disable the hierarchical_namespace?

File: src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_account_scenarios.py (hunk 2 above)
Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_account_scenarios.py]: np

File: src/azure-cli/azure/cli/command_modules/storage/_params.py (hunk 1 above)
Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/_params.py]: > Can we disable the hierarchical_namespace?

Default is disabled.

Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/_params.py]: > What's the type of it?

boolean

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/storage/_params.py]: > 
> 
> > What's the type of it?
> 
> boolean

Do we need a `arg_type=get_three_state_flag()`?

File: src/azure-cli/HISTORY.rst [hunk 3]
```diff
@@ -21,6 +21,10 @@ Release History
 
 * Fix #10996: Polish error for `--force-change-password-next-login` in `az ad user update` when `--password` is not specified
 
+**Storage**
+
+* storage account create: Add --enable-hierarchical-namespace to support filesystem semantics in blob service.
```
Comment (qwordy) [inline src/azure-cli/HISTORY.rst]: filesystem -> file system

File: src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_account_scenarios.py [hunk 4]
```diff
@@ -169,6 +169,27 @@ def test_storage_create_default_sku(self, resource_group):
...
+    def test_storage_create_with_hns(self, resource_group):
+        name = self.create_random_name(prefix='cli', length=24)
+        create_cmd = 'az storage account create -n {} -g {} --kind StorageV2 --hns'.format(name, resource_group)
+        self.cmd(create_cmd, checks=[JMESPathCheck('isHnsEnabled', True)])
+
+    @api_version_constraint(ResourceType.MGMT_STORAGE, min_api='2018-02-01')
+    @ResourceGroupPreparer(location='southcentralus', name_prefix='cli_storage_account_hns')
+    def test_storage_create_with_hns_true(self, resource_group):
+        name = self.create_random_name(prefix='cli', length=24)
+        create_cmd = 'az storage account create -n {} -g {} --kind StorageV2 --hns true'.format(name, resource_group)
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_account_scenarios.py]: one question. Does all kinds of storage support `--hns`?

Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_account_scenarios.py]: No. It only support for StorageV2.

**Pull request merged**